ssr/src/page/post_view/mod.rs

```diff
@@ -222,26 +164,43 @@ pub fn PostViewWithUpdatesMLFeed(initial_post: Option<PostDetails>) -> impl Into
                 let mut cnt = 0;
                 while let Some(chunk) = chunks.next().await {
                     cnt += chunk.len();
-                    video_queue.try_update(|q| {
+                    update!(move |video_queue, priority_q| {
                         for uid in chunk {
                             let uid = try_or_redirect!(uid);
-                            q.push(uid);
+
+                            if video_queue.len() < 10 {
+                                video_queue.push(uid);
+                            } else {
+                                priority_q.push(uid);
+                            }
                         }
                     });
                 }
+
                 leptos::logging::log!("feed type: {:?}", res.res_type);
                 if res.res_type != FeedResultType::MLFeed {
                     fetch_cursor.try_update(|c| {
                         c.set_limit(15);
-                        c.advance_and_set_limit(20)
+                        c.advance_and_set_limit(30)
                     });
                 }
 
                 if res.end || cnt >= 8 {
                     queue_end.try_set(res.end);
-                    break;
                 }
             }
+
+            update!(move |video_queue, priority_q| {
```

Comment:
I was wondering if it was possible to just use the priority queue and remove video queue entirely

Reply:
so priority queue will keep changing right. we want the videos before current_idx to remain same. Also iter() in priroti_queue gives arbitrary order (documented - https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.iter)